modules/setup-systemd-resolved/setup-systemd-resolved

```diff
@@ -67,14 +67,30 @@ function assert_not_empty {
   fi
 }
 
+function has_yum {
+  [ -n "$(command -v yum)" ]
+}
+
+function has_apt_get {
+  [ -n "$(command -v apt-get)" ]
+}
+
 function install_dependencies {
   local -r consul_ip="$1"
 
   log_info "Installing dependencies"
-  sudo apt-get update -y
-  echo iptables-persistent iptables-persistent/autosave_v4 boolean true | sudo debconf-set-selections
-  echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
-  sudo apt-get install -y iptables-persistent
+  if has_apt_get; then
+    sudo apt-get update -y
+    echo iptables-persistent iptables-persistent/autosave_v4 boolean true | sudo debconf-set-selections
+    echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
+    sudo apt-get install -y iptables-persistent
```

Comment:
NIT: Not related to this PR, but I think we want:

```suggestion
    sudo DEBIAN_FRONTEND=noninteractive apt-get install -y iptables-persistent
```

Without it, we may occasionally get an interactive prompt here, causing the build to hang on input.